plugins/experimental/alt_svc/alt_svc.cc

```diff
@@ -0,0 +1,81 @@
+/**
+  @file
+  @brief TODO documentation
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include <atscppapi/GlobalPlugin.h>
+#include <atscppapi/PluginInit.h>
+#include "ip_host_map.h"
+#include "default.h"
+
+using namespace atscppapi;
+using namespace std;
+
+// plugin registration info
+static char vendor_name[]   = "Yahoo! Inc.";
+static char support_email[] = "ats-devel@yahoo-inc.com";
+
+namespace
+{
+GlobalPlugin *plugin;
+}
+
+class AltSvcHeaderPlugin : public GlobalPlugin
+{
+private:
+  IpHostMap *_hostmap;
```

Comment:
`std::unique_ptr` would probably be better here, if this can be changed, and a non-pointer member if not.